src/SamUser/Entity/User.php

```diff
@@ -230,6 +231,44 @@ public function getRoles()
      */
     public function addRole($role)
     {
-        $this->roles[] = $role;
+        $this->roles->add($role);
+    }
+    
+    /**
+     * Remove a role from the user.
+     *
+     * @param Role $role
+     *
+     * @return void
+     */
```

Comment:
Sure, i'll change the foreach block. I usually return the entity (return $this;) on set / add & remove. Shall I put that in or just remove the "return void"?